src/management-system-v2/lib/process-export/copy-process-image.ts

```diff
@@ -1,34 +1,100 @@
-import { toBpmnObject, toBpmnXml } from '@proceed/bpmn-helper';
-
 import { getSVGFromBPMN } from './util';
 import { getPNGFromSVG } from './image-export';
+import { BPMNCanvasRef } from '@/components/bpmn-canvas';
+import Modeler from 'bpmn-js/lib/Modeler';
+import NavigatedViewer from 'bpmn-js/lib/NavigatedViewer';
+
+async function getRootElementAndSelection(modeler: BPMNCanvasRef | Modeler | NavigatedViewer) {
```

Comment:
I would just transform this to a getPNG function since you the two steps that you currently do after this function is called are still the same in the two calling functions and the logic for getting the xml in the one function is also included in the other function so just directly call this function at the start of the other two functions to get the PNG and then run the logic that differs